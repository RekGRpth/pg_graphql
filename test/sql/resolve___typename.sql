begin;

    create table account(
        id int primary key
    );


    insert into public.account(id)
    values
        (1);


    select jsonb_pretty(
        graphql.resolve($$
    {
      accountCollection {
        __typename
        pageInfo {
          __typename
        }
        edges {
          __typename
          node {
            __typename
          }
        }
      }
    }
        $$)
    );


    select graphql.resolve($$
    {
      account(id: "WyJhY2NvdW50IiwgMV0=") {
        __typename
      }
    }
    $$);

rollback;
