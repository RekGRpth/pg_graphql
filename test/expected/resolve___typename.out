begin;
    create table account(
        id int primary key
    );
    insert into public.account(id)
    values
        (1);
    select jsonb_pretty(
        graphql.resolve($$
    {
      allAccounts {
        __typename
        pageInfo {
          __typename
        }
        edges {
          __typename
          node {
            __typename
          }
        }
      }
    }
        $$)
    );
                  jsonb_pretty                   
-------------------------------------------------
 {                                              +
     "data": {                                  +
         "allAccounts": {                       +
             "edges": [                         +
                 {                              +
                     "node": {                  +
                         "__typename": "Account"+
                     },                         +
                     "__typename": "AccountEdge"+
                 }                              +
             ],                                 +
             "pageInfo": {                      +
                 "__typename": "PageInfo"       +
             },                                 +
             "__typename": "AccountConnection"  +
         }                                      +
     },                                         +
     "errors": [                                +
     ]                                          +
 }
(1 row)

    select graphql.resolve($$
    {
      account(id: "WyJhY2NvdW50IiwgMV0=") {
        __typename
      }
    }
    $$);
                            resolve                             
----------------------------------------------------------------
 {"data": {"account": {"__typename": "Account"}}, "errors": []}
(1 row)

rollback;
